spec/workflow/choice_rule_spec.rb

```diff
@@ -147,7 +147,7 @@
       end
 
       context "with an invalid compare key" do
-        let(:choices) { [{"Variable" => "$.foo", "InvalidCompare" => "$.bar", "Next" => "FirstMatchState"}] }
+        let(:choices) { [{"Variable" => "$.foo", "StringBeGone" => "$.bar", "Next" => "FirstMatchState"}] }
```

Comment:
Before this fix, `String.*` would pass validation.

`InvalidCompare` will fail in the new code as well, but I changed this to show the new improvement.